base_export_email/ir_actions.py

```diff
@@ -0,0 +1,157 @@
+#    Service Company.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import orm, fields
+from openerp.tools.translate import _
+from openerp.addons.web.controllers import main
+from openerp.tools.safe_eval import safe_eval
+import time
+import base64
+
+
+class actions_server(orm.Model):
+    _inherit = "ir.actions.server"
+
+    def __init__(self, pool, cr):
+        super(actions_server, self).__init__(pool, cr)
+        # add state 'export_email'
+        self._columns['state'].selection.append(('export_email',
+                                                 _('Export data by email')))
+        return
+
+    _columns = {'model': fields.related('model_id', 'model',
+                                        type="char", size=256, string='Model'),
+                'filter_id': fields.many2one(
+                    'ir.filters', string='Filter', ondelete='restrict'),
+                'email_template_id': fields.many2one(
+                    'email.template', string='Template', ondelete='restrict'),
+                'saved_export_id': fields.many2one(
+                    'ir.exports', string='Saved export', ondelete='restrict'),
+                'fields_to_export': fields.text(
+                    'Fields to export',
+                    help="The list of fields to export. \
+                          The format is ['name', 'seller_ids/product_name']")
+                }
+
+    _defaults = {'fields_to_export': '[]'}
+
+    def onchange_model_id(self, cr, uid, ids, model_id, context=None):
+        """
+        Used to set correct domain on filter_id and saved_export_id
+        """
+        data = {'model': False,
+                'filter_id': False,
+                'email_template_id': False,
+                'saved_export_id': False}
+        if model_id:
+            model = self.pool['ir.model'].browse(cr, uid, model_id,
+                                                 context=context)
+            data.update({'model': model.model})
+        return {'value': data}
+
+    def _search_data(self, cr, uid, action, context=None):
+        """
+         search data to export
+        """
+        obj_pool = self.pool[action.model]
```

Comment:
Please elaborate the docstring, or remove it. A docstring that does not say anything more than the method name is useless.